libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -5,27 +5,39 @@ import {
   clearSearch,
   getAllBooks,
   ReadingListBook,
-  searchBooks
+  searchBooks,
+  removeFromReadingList
 } from '@tmo/books/data-access';
+import {Observable,  of} from 'rxjs';
+import {debounceTime,tap,switchMap,filter, map, startWith, distinctUntilChanged} from 'rxjs/operators';
 import { FormBuilder } from '@angular/forms';
-import { Book } from '@tmo/shared/models';
+
+import { Book,ReadingListItem } from '@tmo/shared/models';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
 import { Subscription } from 'rxjs';
 
+
 @Component({
   selector: 'tmo-book-search',
   templateUrl: './book-search.component.html',
   styleUrls: ['./book-search.component.scss']
 })
 export class BookSearchComponent implements OnInit,OnDestroy {
   books: ReadingListBook[];
+
+
   subscription : Subscription;
+
   searchForm = this.fb.group({
     term: ''
   });
-
+  item : ReadingListItem;
```

Comment:
Better to avoid storing this as a class property. Having transient class properties can make the code hard to understand.

You already have enough information from the existing action:

```
this.store.dispatch(addToReadingList({ book }));
```

Reply:
changed accordingly